vulcanai2/tests/models/test_layers.py

```diff
@@ -1,42 +1,196 @@
-            kernel_size=(5,5)
+            kernel_size=(5, 5)
         )
-    
+
     def test_forward(self, conv_unit):
         """Confirm size is expected after forward."""
         test_input = torch.ones([1, conv_unit.in_channels, 28, 28])
         output = conv_unit.forward(test_input)
         # No padding with 2 5x5 kernels leads from 28x28 -> 24x24
         assert output.size() == torch.ones([1, conv_unit.out_channels, 24, 24]).size()
 
+
+class TestSeluInit:
+    """To test selu initialized layer properties."""
+
+    @pytest.fixture
+    def dense_unit(self):
+        """Create a dense unit fixture."""
+        return DenseUnit(
+            in_features=10,
+            out_features=10
+        )
+
+    @pytest.fixture
+    def conv_unit(self):
+        """Create a conv unit fixture."""
+        return ConvUnit(
+            conv_dim=2,
+            in_channels=10,
+            out_channels=10,
+            kernel_size=(5, 5)
+        )
+
+    def test_dense_selu_weight_change(self, dense_unit):
+        """Confirm SELU weight init properties hold for dense net."""
+        starting_weight = copy.deepcopy(dense_unit._kernel.weight)
+        std = round(math.sqrt(1. / 10), 1)
+
+        dense_unit.weight_init = selu_weight_init_
+        dense_unit._init_weights()
+        new_weight = copy.deepcopy(dense_unit._kernel.weight)
+        assert (torch.equal(starting_weight, new_weight) is False)
+        assert pytest.approx(round(new_weight.std().item(), 1), 0.1) == std
+        assert (int(new_weight.mean().item()) == 0.0)
+
+    def test_conv_selu_weight_change(self, conv_unit):
+        """Confirm SELU weight init properties hold for conv net."""
+        starting_weight = copy.deepcopy(conv_unit._kernel.weight)
+        std = round(math.sqrt(1. / 250), 1)
+        conv_unit.weight_init = selu_weight_init_
+        conv_unit._init_weights()
+        new_weight = copy.deepcopy(conv_unit._kernel.weight)
+        assert (torch.equal(starting_weight, new_weight) is False)
+        assert pytest.approx(round(new_weight.std().item(), 1), 0.1) == std
+        assert (int(new_weight.mean().item()) == 0)
+
+    def test_dense_selu_bias_change(self, dense_unit):
+        """Confirm SELU bias init properties hold for dense net."""
+        starting_bias = copy.deepcopy(dense_unit._kernel.bias)
+
+        dense_unit.bias_init = selu_bias_init_
+        dense_unit._init_bias()
+        new_bias = copy.deepcopy(dense_unit._kernel.bias)
+        assert (torch.equal(starting_bias, new_bias) is False)
+        assert (round(new_bias.std().item(), 1) == 0.0)
+        assert (int(new_bias.mean().item()) == 0)
+
+    def test_conv_selu_bias_change(self, conv_unit):
+        """Confirm SELU bias init properties hold for conv net."""
+        starting_bias = copy.deepcopy(conv_unit._kernel.bias)
+
+        conv_unit.bias_init = selu_bias_init_
+        conv_unit._init_bias()
+        new_bias = copy.deepcopy(conv_unit._kernel.bias)
```

Comment:
same comment on the deepcopy

Reply:
@rfratila Just pushed update that removed copy